hs_core/views/__init__.py

```diff
@@ -697,3 +697,18 @@ def _set_resource_sharing_status(request, user, resource, is_public, is_discover
         # set isPublic metadata AVU accordingly
         istorage = IrodsStorage()
         istorage.setAVU(resource.short_id, "isPublic", str(resource.raccess.public))
+
+
+def _get_message_for_setting_resource_flag(resource, resource_flag):
```

Comment:
the two checks, resource.has_required_content_files() and resource.metadata.has_all_required_elements(), were already called in resource.can_be_public in the calling method _set_resource_sharing_status(), then are called again in order to set more informative message. I think there is a better way to remove the duplicate checks as well as duplicate code. For example, if can_be_public() method can return true or false as well as has_files and has_metadata flags, you can directly set up message accordingly without even needing this _get_message_for_setting_resource_flag() method.